src/hotspot/cpu/s390/macroAssembler_s390.cpp

```diff
@@ -3671,17 +3671,14 @@ void MacroAssembler::verify_secondary_supers_table(Register r_sub_klass,
                                                    Register r_temp1,
                                                    Register r_temp2,
                                                    Register r_temp3) {
-  assert_different_registers(r_sub_klass, r_super_klass, r_result, r_temp1, r_temp2, r_temp3);
+  assert_different_registers(r_sub_klass, r_super_klass, r_result, r_temp1, r_temp2, r_temp3, Z_R0_scratch);
```

Comment:
```suggestion
  assert_different_registers(r_sub_klass, r_super_klass, r_result, r_temp1, r_temp2, r_temp3);
```

Reply:
But we are still using Z_R0, to resize the frame, at line 3708 and 3718.

Reply:
I missed that because you were using different names for the same register in this function. I searched for `Z_R0_scratch`. It would be better if you decided what to call `Z_R0`, and used the same name consistently for all usages, including `assert_different_registers()`.

Reply:
I have moved the result in volatile-float registers, instead of using stack for the shuffling. With that we can get rid of `Z_R0_scratch` from `assert_different_registers` as per your suggestion.

Reply:
> I have moved the result in volatile-float registers, instead of using stack for the shuffling. With that we can get rid of `Z_R0_scratch` from `assert_different_registers` as per your suggestion.

Fair enough, but this is a general recommendation to do with good practice in hand-coded assembler. Using aliases for registers can result in a maintenance programmer not noticing that two different names refer to the same thing. , This has led to bugs that aren't revealed in tests but crash in production. Sure, aliases (e.g `obj`, `klass`, `index` etc. can help readbility, but be careful.

This isn't intended to be a total ban, but a note of caution.